lib/fluent/plugin/in_unix_client.rb

```diff
@@ -160,14 +174,42 @@ def extract_records
 
         pos_read = 0
         while pos_next_delimiter = @buf.index(@delimiter, pos_read)
-          records << @buf[pos_read...pos_next_delimiter]
+          fixed = fix_format(@buf[pos_read...pos_next_delimiter])
+          records << fixed unless fixed.empty?
           pos_read = pos_next_delimiter + @delimiter.size
         end
 
         @buf.slice!(0, pos_read) if pos_read > 0
 
         records
       end
+
+      private
+
+      def fix_format(record)
+        return record if record.empty?
+        return record unless @for_json
+
+        fix_for_json_list_structure(record)
+      end
+
+      def fix_for_json_list_structure(record)
+        if record[0] == "[".freeze && record[-1] == "]".freeze
```

Comment:
Validating by `JSON.parse` might be more robust against various JSON strings like this:

https://github.com/fluent/fluentd/blob/b9203165182be15e60574ec6f1ce78df0894b304/lib/fluent/config/literal_parser.rb#L271-L275

```ruby
          begin
            result = JSON.parse(buffer + line_buffer)
          rescue JSON::ParserError
            # Incomplete json string yet
          end
```

The side effect of this way is it might decrease the performance.

Reply:
Thank you for the advice!

> The side effect of this way is it might decrease the performance.

I don't know in detail how the JSON list format is usually used.
So I was worried about the case where the letter `]` was not returned for a while.

In most cases, can this side effect be ignored?

Reply:
> So I was worried about the case where the letter `]` was not returned for a while.

I misunderstood. We can use `JSON.parse` just to validate the format.
I fixed this and made this feature optional.

Thanks!